tmt/steps/provision/mock.py

```diff
@@ -0,0 +1,552 @@
+import fcntl
+import io
+import os
+import select
+import shlex
+import subprocess
+from typing import Any, Callable, Optional, Union
+
+import tmt
+import tmt.base
+import tmt.log
+import tmt.steps
+import tmt.steps.provision
+import tmt.utils
+from tmt._compat.typing import Self
+from tmt.container import container, field
+from tmt.utils import Command, OnProcessEndCallback, OnProcessStartCallback, Path, ShellScript
+from tmt.utils.wait import Waiting
+
+
+@container
+class MockGuestData(tmt.steps.provision.GuestData):
+    root: Optional[str] = field(
+        default=None,
+        option=('-r', '--root'),
+        metavar='ROOT',
+        help="""
+             Mock chroot configuration file.
+             The `--root` flag to be passed to the mock process.
+             """,
+    )
+    rootdir: Optional[str] = field(
+        default=None,
+        option=('--rootdir'),
+        metavar='ROOTDIR',
+        help='The path for where the chroot should be built.',
+    )
+
+
+@container
+class ProvisionMockData(MockGuestData, tmt.steps.provision.ProvisionStepData):
+    pass
+
+
+class MockShell:
+    # Lazy object that collects chunks of bytes and decodes them when a
+    # newline is encountered
+    class Stream:
+        def __init__(self, logger: Callable[[str], None]):
+            self.logger = logger
+            self.output = b''
+            self.string = ''
+
+        def __iadd__(self, content: bytes) -> Self:
+            self.output += content
+            while True:
+                pos = self.output.find(b'\n')
+                if pos == -1:
+                    break
+                string = self.output[:pos].decode('utf-8', errors='replace')
+                self.output = self.output[pos + 1 :]
+                self.logger(string)
+                self.string += string
+                self.string += '\n'
+            return self
+
+    def __init__(self, parent: 'GuestMock', root: Optional[str], rootdir: Optional[str]):
+        self.parent = parent
+        self.root = root
+        self.rootdir = rootdir
+        self.mock_shell: Optional[subprocess.Popen[str]] = None
+        self.epoll: Optional[select.epoll] = None
+        # Required by loggers
+        self.pid: Optional[int] = None
+
+        self.command_prefix = Command('mock')
+        if self.root is not None:
+            self.command_prefix += ['-r', self.root]
+        if self.rootdir is not None:
+            self.command_prefix += ['--rootdir', self.rootdir]
+
+        root_path = (
+            (self.command_prefix + ['--print-root-path'])
```

Comment:
```suggestion
            (self.command_prefix + Command('--print-root-path'))
```
I believe this is the common standard all across. Similarly
```python
self.command_prefix += Command('-r', self.root)
```